src/platforms/esp32/components/avm_builtins/network_driver.c

```diff
@@ -848,8 +867,32 @@ Context *network_driver_create_port(GlobalContext *global, term opts)
     return ctx;
 }
 
+//
+// Destructor
+//
+
+void network_driver_stop(GlobalContext *global)
+{
+    wifi_mode_t mode;
+    esp_err_t err = esp_wifi_get_mode(&mode);
+    if ((err == ESP_OK) && ((mode == WIFI_MODE_STA) || (mode == WIFI_MODE_APSTA))) {
+        err = esp_wifi_disconnect();
+        if (UNLIKELY(err != ESP_OK)) {
+            ESP_LOGE(TAG, "Error while disconnecting from AP (error: %i)", err);
+        }
+    }
+    err = esp_wifi_stop();
+    if (UNLIKELY(err != ESP_OK)) {
+        ESP_LOGE(TAG, "Error while stopping the wifi driver (error: %i)", err);
+    }
+    err = esp_wifi_deinit();
+    if (UNLIKELY(err != ESP_OK)) {
+        ESP_LOGE(TAG, "Error releasing wifi driver resources (error: %i)", err);
+    }
+}
+
 #ifdef CONFIG_AVM_ENABLE_NETWORK_PORT_DRIVER
 
-REGISTER_PORT_DRIVER(network, network_driver_init, NULL, network_driver_create_port)
+REGISTER_PORT_DRIVER(network, network_driver_init, network_driver_stop, network_driver_create_port)
```

Comment:
`network_driver_stop` will not be called when calling `network:stop/0`. It gets called during tests (when GlobalContext is destoryed and recreated).
So this change sounds correct, but it doesn't do what you are saying.